roles/postgresql-users/tasks/main.yml

```
@@ -21,4 +21,20 @@
   when: postgresql_users is defined and postgresql_users | length > 0
   tags: postgresql_users
 
+- name: Grant roles to users
+  community.postgresql.postgresql_membership:
+    group: "{{ item.role }}"
+    target_role: "{{ item.name }}"
+    login_host: "127.0.0.1"
+    login_port: "{{ postgresql_port }}"
+    login_user: "{{ patroni_superuser_username }}"
+    login_password: "{{ patroni_superuser_password }}"
+    state: present
+  ignore_errors: true
+  loop: "{{ postgresql_users | flatten(1) }}"
+  loop_control:
+    label: "{{ item.name }}"
+  when: postgresql_users is defined and postgresql_users | length > 0
```

```suggestion
  when:
    - postgresql_users is defined
    - postgresql_users | length > 0
    - item.role | default('') | length > 0
```

If the 'role' is defined, but empty, then community.postgresql.postgresql_membership may not work correctly or give an error, since an empty value for the role may be invalid.